src/Fields/abstract_field.jl

```diff
@@ -30,11 +31,34 @@ function validate_field_data(X, Y, Z, data, grid)
 end
 
 #####
-##### AbstractField functionality
+##### Computing AbstractField
 #####
 
-# Overload compute! for custom fields to produce non-default behavior
-compute!(f::AbstractField) = nothing
+# Note: overload compute! for custom fields to produce non-default behavior
+
+"""
+    compute!(field)
+
+Computes `field.data`.
+"""
+compute!(field::AbstractField) = nothing
+
+"""
+    @compute(exprs...)
+
+Call compute! on fields after defining them.
+"""
+macro compute(def)
+    expr = Expr(:block)
+    field = def.args[1]
+    push!(expr.args, :($(esc(def))))
+    push!(expr.args, :(compute!($(esc(field)))))
+    return expr
+end
```

Comment:
Kinda against defining this macro but we already discussed this on Slack. It seems appropriate for post-processing, which should become more possible with the fields abstraction so let's keep it.

Reply:
We can open an issue to remove it after the PR is merged?

Reply:
Nah we should keep it and see how it gets used. It's already useful in the tests. I might just have a personal prejudice against macros.